### PR TITLE
MMCA-4777: Upgrade scala and library dependancies to latest version - 2024

### PR DESCRIPTION
**Pull request by HariHmrc**

### CODE REVIEW

File: app/connectors/SdesConnector.scala [hunk 1]
```diff
@@ -75,18 +76,16 @@ class SdesConnector @Inject()(httpClient: HttpClient,
...
-                                             readSeq: HttpReads[Seq[A]]): Future[Seq[B]] = {
+                                            (implicit readSeq: HttpReads[Seq[A]]): Future[Seq[B]] = {
     metricsReporterService.withResponseTimeLogging(metricsName) {
-      httpClient.GET[HttpResponse](
-        url,
-        headers = Seq("x-client-id" -> appConfig.xClientIdHeader, "X-SDES-Key" -> key)
-      )(reads, HeaderCarrier(), implicitly)
-        .map(readSeq.read("GET", url, _))
+      httpClient.get(url"$urlLink")(HeaderCarrier())
+        .setHeader(("x-client-id" -> appConfig.xClientIdHeader), ("X-SDES-Key" -> key))
```
Comment (magicaljellybeans) [inline app/connectors/SdesConnector.scala:86]: Could we move the Header keys in a separate object please? We do this in other frontend repos.

**Review by magicaljellybeans — COMMENTED** [review]
Just one comment Hari.

File: app/connectors/SdesConnector.scala [hunk 2]
```diff
@@ -76,14 +76,17 @@ class SdesConnector @Inject()(httpClient: HttpClientV2,
     )
   }
 
+  private def addXHeaders(hc: HeaderCarrier, key: String): HeaderCarrier =
```
Comment (magicaljellybeans) [inline app/connectors/SdesConnector.scala:79]: I meant like this: https://github.com/hmrc/customs-manage-authorities-frontend/blob/main/app/config/Headers.scala

Sorry, should have made that clearer.

**Review by magicaljellybeans — APPROVED** [review]

**Review by AdamCoops — APPROVED** [review]
lgtm

**Pull request merged**